### PR TITLE
Gurukul Support & supporting non-session Portal links & token auth

### PR DESCRIPTION
**Pull request by sudheshna-donthineni**
- Added query parameters
- Changed variable names
- Lifted certain computed methods from child to parent (`NewSignIn` to `Home`)
- Also, stores access token and refresh token in cookies

### CODE REVIEW

File: src/pages/Home.vue [hunk 1]
```diff
@@ -175,32 +268,98 @@ export default {
...
+        (this.auth_type && this.auth_type.split(",")) ||
+        (this.groupData && this.groupData.input_schema.auth_type.split(","))
+      );
+    },
+
+    /** Returns if sign up flow should be enabled */
+    isSignupEnabled() {
+      return (
+        (this.sessionData && this.sessionData.enable_signup == "True") ||
+        this.enable_signup == "True"
```
Comment (suryabulusu) [inline src/pages/Home.vue:277]: when will this.enable_signup = True happen? its not being set anywhere?

File: src/pages/NewSignup.vue [hunk 2]
```diff
@@ -300,16 +305,16 @@ export default {
...
       ) {
         sendSQSMessage(
           "attendance-sign-up",
           this.$store.state.sessionData.purpose["sub-type"],
-          this.$store.state.sessionData.platform,
-          this.$store.state.sessionData.platform_id,
+          this.platform,
+          this.platform_id,
           this.userData["user_id"],
           "",
```
Comment (suryabulusu) [inline src/pages/NewSignup.vue:295]: what is this for? can u add besides it in comment

File: src/pages/Home.vue (hunk 1 above)
Comment (sudheshna-donthineni) [reply src/pages/Home.vue:277]: Its a URL parameter.

File: src/pages/NewSignup.vue (hunk 2 above)
Comment (sudheshna-donthineni) [reply src/pages/NewSignup.vue:295]: Done

**Pull request merged**